cmake/BuildLibCURL.cmake

```diff
@@ -29,6 +29,7 @@ macro(curl_build)
 
     # Switch on the static build.
     list(APPEND LIBCURL_CMAKE_FLAGS "-DBUILD_STATIC_LIBS=ON")
+    list(APPEND LIBCURL_CMAKE_FLAGS "-DBUILD_STATIC_CURL=ON")
```

Comment:
I would add a comment in the code that clarifies a meaning of the option: it is easy to wrongly understand it as enabling of building the executable file (which is controlled by `BUILD_CURL_EXE`).

Something like the following.

```cmake
# Switch on static build.
#
# The BUILD_STATIC_CURL option has no effect, because
# BUILD_CURL_EXE is OFF. It is set for consistency:
# our approach is to set all the options explicitly.
list(APPEND LIBCURL_CMAKE_FLAGS "-DBUILD_STATIC_LIBS=ON")
list(APPEND LIBCURL_CMAKE_FLAGS "-DBUILD_STATIC_CURL=ON")
```

Reply:
Added comment similar to your example.